server/permissions/UserAuth.ts

```diff
@@ -46,29 +43,41 @@ const SUPER_ROLES = [UserRole.Host, UserRole.Developer];
 
 export class UserAuthWithContext extends UserAuth {
   private context: PermissionContext;
-  private hasSuperRole: boolean = false;
+  private superRoleLoaded: boolean = false;
+  private userHasSuperRole: boolean = false;
 
   constructor(user: IUser, context: PermissionContext) {
     super(user);
     this.context = context;
-    this.hasSuperRole =
-      this.user.admin || SUPER_ROLES.some((role) => this.context.hasRole(this.user, role));
+  }
+
+  private async hasSuperRole() {
+    if (this.superRoleLoaded && this.userHasSuperRole) return true;
+    if (this.user?.admin) {
+      this.userHasSuperRole = true;
+    } else {
+      this.userHasSuperRole = await Promise.all(
+        SUPER_ROLES.map((role) => this.context.hasRole(this.user, role))
+      ).then((results) => results.some((result) => result));
+    }
+    return this.userHasSuperRole;
```

Comment:
superRoleLoaded is never set?

Reply:
Oops missed it, fixed in fe1afdc4219d5e6f1062cca2aaac9e2ba69169fb